src/server/getters/get-categories.ts

```diff
@@ -42,32 +53,84 @@ async function getCategoriesPopularity(db: Knex, universe: string): Promise<Arra
 }
 
-function convertTagsToArray(tagRows: Array<TagRow>): TagCount {
-  return _.chain(tagRows).map((t) => [(t.tag1 || "").toUpperCase(), (t.tag2 || "").toUpperCase()])
-    .flatten()
-    .filter((tag) => tag !== "")
-    .countBy()
-    .value();
-}
+// function convertTagsToArray(tagRows: Array<TagRow>): TagCount {
+//   return _.chain(tagRows).map((t) => [(t.tag1 || "").toUpperCase(), (t.tag2 || "").toUpperCase()])
+//     .flatten()
+//     .filter((tag) => tag !== "")
+//     .countBy()
+//     .value();
+// }
 
-async function getTagsCountByCategory(db: Knex, universe: string): Promise<{ [category: string]: TagCount }> {
-  const tagsRows: Array<TagRow> = await db.select(["tag1", "tag2", "category"]).from("markets").where({ universe });
-  return _.chain(tagsRows)
-    .groupBy((tagRow) => tagRow.category.toUpperCase())
-    .mapValues(convertTagsToArray)
-    .value();
-}
+// s/getTagsForCategory ?
+// async function getTagsCountByCategory(db: Knex, universe: string): Promise<{ [category: string]: TagCount }> {
+//   const tagsRows: Array<TagRow> = await db.select(["tag1", "tag2", "category"]).from("markets").where({ universe });
+//   return _.chain(tagsRows)
+//     // is this `groupBy(...): Map<category, [tag1, tag2, category]>` and incorrectly treat the categories as tags? --> no, because the input is an object and the keys tag1/tag2 are extracted in convertTagsToArray(), discarding the category
+//     .groupBy((tagRow) => tagRow.category.toUpperCase())
+//     .mapValues(convertTagsToArray)
+//     .value();
+// }
 
-export async function getCategories(db: Knex, augur: Augur, params: t.TypeOf<typeof CategoriesParams>): Promise<Array<UICategory>> {
+// TODO post-condition: categories.map(_.category.toUpperCase()).unique.length == categories.length --> ie. everything is uppercased (or titled cased)
+export async function getCategories(db: Knex, augur: Augur, params: t.TypeOf<typeof CategoriesParams>): Promise<Array<UICategory<string>>> { // we use UICategory<string> to be consistent with the pre-existing UIMarketInfo<string> in get-markets.ts
   const universeInfo = await db.first(["universe"]).from("universes").where({ universe: params.universe });
   if (universeInfo === undefined) throw new Error(`Universe ${params.universe} does not exist`);
-  const tagCountByCategory = await getTagsCountByCategory(db, params.universe);
-  const categoriesResponse = await getCategoriesPopularity(db, params.universe);
-  return _.map(categoriesResponse, (categoryResponse) => {
-    const category = categoryResponse.category;
-    return Object.assign(
-      { tags: tagCountByCategory[category] || [] },
-      categoryResponse,
-    );
-  });
+  // const tagCountByCategory = await getTagsCountByCategory(db, params.universe);
+  // const categoriesResponse = await getCategoriesPopularity(db, params.universe);
+  return Promise.resolve([
+    {
+      categoryName: "foo",
+      nonFinalizedOpenInterest: "39383",
+      tags: [
```

Comment:
Assumed unsorted?

Reply:
Yes, the categories/tags are unsorted in AN and are sorted on the front end.